docs/Middleware.md

```diff
@@ -0,0 +1,261 @@
+            other entry in the group</td>
+        </tr>
+    </tbody>
+</table>
+
+### Defining and registering middleware components
+
+A piece of Kettle middleware is derived from grade `kettle.middleware`. This is a very simple grade which defines a single invoker named `handle` which accepts one argument, a `kettle.request`, and returns a 
+promise representing the completion of the middleware. Conveniently a `fluid.promise` implementation is available in the framework, but you can return any variety of `thenable` that you please. Here is a skeleton,
+manually implemented middleware component:
+
+```javascript
+fluid.defaults("examples.customMiddleware", {
+    gradeNames: "kettle.middleware",
+    invokers: {
+        handle: "examples.customMiddleware.handle"
+    }
+});
+
+examples.customMiddleware.handle = function (request) {
+    var togo = fluid.promise();
+    if (request.req.params.id === 42) {
+        togo.resolve();
+    } else {
+        togo.reject({
+            isError: true,
+            statusCode: 401,
+            message: "Only the id 42 is authorised"
+        });
+    }
+    return togo;
+};
+```
+
+The framework makes it very easy to adapt any standard express middleware into a middleware component by means of the adaptor grade `kettle.plainMiddleware`. This accepts any standard express
+middleware as the option named `middleware` and from it fabricates a `handle` method with the semantic we just saw earlier. Any options that the middleware accepts can be forwarded to it from
+the component's options. Here is an example from the framework's own `json` middleware grade:
+
+```javascript
+kettle.npm.bodyParser = require("body-parser");
+
+fluid.defaults("kettle.middleware.json", {
+    gradeNames: ["kettle.plainMiddleware"],
+    middlewareOptions: {}, // see https://github.com/expressjs/body-parser#bodyparserjsonoptions
+    middleware: "@expand:kettle.npm.bodyParser.json({that}.options.middlewareOptions)"
+});
+```
+
+Consult the Infusion documentation on the [compact format for expanders](http://docs.fluidproject.org/infusion/development/ExpansionOfComponentOptions.html#compact-format-for-expanders) if you
+are unfamiliar with this syntax for designating elements in component options which arise from function calls.
+
+### Built-in standard middleware bundled with Kettle
+
+Here we describe the built-in middleware supplied with Kettle, which is mostly sourced from standard middleware in the [express](http://expressjs.com/) and [pillarjs](https://github.com/pillarjs)
+communities. You can consult the straightforward implementations in [KettleMiddleware.js](https://github.com/fluid-project/kettle/tree/master/lib/KettleMiddleware.js) for suggestions for how
+to implement your own.
+
+<div style="font-size: smaller">
+<table>
```

Comment:
I was unable to find a way to lay it out better, after much tinkering. Any suggestions enthusiastically received!